src/Illuminate/Macroable/Traits/Macroable.php

```diff
@@ -31,14 +31,16 @@ public static function macro($name, $macro)
     /**
      * Mix another object into the class.
      *
-     * @param  object  $mixin
+     * @param  class-string|object  $mixin
      * @param  bool  $replace
      * @return void
      *
      * @throws \ReflectionException
      */
     public static function mixin($mixin, $replace = true)
     {
+        $mixin = is_string($mixin) ? new $mixin() : $mixin;
```

Comment:
Might be nice to build the mixin from the container so dependencies can be injected

Reply:
That thought did cross my mind, implemented